Google.Api.Generator/Generation/MethodDetails.cs

```diff
@@ -269,11 +271,50 @@ public Signature(ServiceDetails svc, MessageDescriptor msg , string sig)
             public bool HasDeprecatedField => Fields.Any(field => field.IsDeprecated);
         }
 
+        /// <summary>
+        /// Represents a routing header model, with multiple ways to match-and-extract
+        /// the value to be evaluated
+        /// </summary>
         public sealed class RoutingHeader
         {
-            public RoutingHeader(string encodedName, IEnumerable<FieldDescriptor> fields) => (EncodedName, Fields) = (encodedName, fields);
-            public string EncodedName { get; }
-            public IEnumerable<FieldDescriptor> Fields { get; }
+            public string EncodedName { get; set; }
+            public List<FieldExtraction> Extractions { get; set; }
+
+            /// <summary>
+            /// Whether a routing header should a full value of a single field with no Regex matching needed.
+            /// True e.g. for all implicit headers, and for the explicit headers where the field pattern is `**`.
+            /// </summary>
+            public bool FullFieldNoRegex => Extractions.Count == 1 && Extractions.Single().NoRegexMatchingNeeded;
+            
+            /// <summary>
+            /// A way to match-and-extract the value from a request's field.
+            /// </summary>
+            public sealed class FieldExtraction
+            {
+                public IEnumerable<FieldDescriptor> Fields { get; set; }
+                public string RegexStr { get; set; }
+                public bool NoRegexMatchingNeeded { get; set; }
+            }
+
+            public static RoutingHeader CreateOneFieldNoRegex(string encodedName, IEnumerable<FieldDescriptor> fields) =>
+                new()
```

Comment:
Note: we're planning on upgrading to the .NET 6 SDK as part of the GAX upgrade, so we could do the same thing here and use features up to and including C# 10.

Reply:
Probably not worth it yet?

Reply:
This was more of a "note for the future" than a proposed change now :)